pkg/cli/image/extract/extract.go

```diff
@@ -593,22 +596,32 @@ func (n *copyFromDirectory) Alter(hdr *tar.Header) (bool, error) {
 type copyFromPattern struct {
 	Base string
 	Name string
+	RootDir bool
 }
 
-func newCopyFromPattern(dir, name string) archive.AlterHeader {
+func newCopyFromPattern(dir, name string, rootDir bool) archive.AlterHeader {
+	if rootDir {
+		return &copyFromPattern{Name: name, RootDir: true}
+	}
 	if !strings.HasSuffix(dir, "/") {
 		dir = dir + "/"
 	}
-	return &copyFromPattern{Base: dir, Name: name}
+	return &copyFromPattern{Base: dir, Name: name, RootDir: false}
 }
 
 func (n *copyFromPattern) Alter(hdr *tar.Header) (bool, error) {
-	if !changeTarEntryParent(hdr, n.Base) {
-		return false, nil
-	}
 	matchName := hdr.Name
-	if i := strings.Index(matchName, "/"); i != -1 {
-		matchName = matchName[:i]
+	if n.RootDir {
+		if !changeTarEntryName(hdr, n.Name) {
+			return false, nil
+		}
+	} else {
+		if !changeTarEntryParent(hdr, n.Base) {
+			return false, nil
+		}
```

Comment:
and here, also, add  `matchName = hdr.Name`

Reply:
(so the fn should be this - it was my bad - otherwise extract from non-root dir is broken bc matchName set b4 hdr.Name update)
```
func (n *copyFromPattern) Alter(hdr *tar.Header) (bool, error) {
    var matchName string
    if n.RootDir {
        if !changeTarEntryName(hdr, n.Name) {
            return false, nil 
        }
        matchName = hdr.Name
    } else {
        if !changeTarEntryParent(hdr, n.Base) {
            return false, nil 
        }
        matchName = hdr.Name
        if i := strings.Index(matchName, "/"); i != -1 {
            matchName = matchName[:i]
        }
    }   
    if ok, err := path.Match(n.Name, matchName); !ok || err != nil {
        klog.V(5).Infof("Excluded %s due to filter %s", hdr.Name, n.Name)
        return false, err 
    }   
    return true, nil 
}
```

Reply:
:) Thank you